superset/models/core.py

```diff
@@ -885,36 +885,67 @@ def wrapper(*args, **kwargs):
             if g.user:
                 user_id = g.user.get_id()
             d = request.form.to_dict() or {}
+
             # request parameters can overwrite post body
             request_params = request.args.to_dict()
             d.update(request_params)
             d.update(kwargs)
+
             slice_id = d.get('slice_id')
+            dashboard_id = d.get('dashboard_id')
 
             try:
                 slice_id = int(
                     slice_id or json.loads(d.get('form_data')).get('slice_id'))
             except (ValueError, TypeError):
                 slice_id = 0
 
+            stats_logger.incr(f.__name__)
+            value = f(*args, **kwargs)
+            sesh = db.session()
+
+            logs = []
+            duration_ms = (datetime.now() - start_dttm).total_seconds() * 1000
+            referrer = request.referrer[:1000] if request.referrer else None
+
             params = ''
             try:
                 params = json.dumps(d)
             except Exception:
                 pass
-            stats_logger.incr(f.__name__)
-            value = f(*args, **kwargs)
-            sesh = db.session()
-            log = cls(
-                action=f.__name__,
-                json=params,
-                dashboard_id=d.get('dashboard_id'),
-                slice_id=slice_id,
-                duration_ms=(
-                    datetime.now() - start_dttm).total_seconds() * 1000,
-                referrer=request.referrer[:1000] if request.referrer else None,
-                user_id=user_id)
-            sesh.add(log)
+
+            # bulk insert
+            explode_by = d.get('explode', None)
+            events = None
+            try:
+                events = json.loads(d.get(explode_by))
+            except Exception:
+                pass
+
+            if isinstance(events, list):
```

Comment:
If you change line #923 to be `events = [d]` you don't need to repeat the `cls(...)` call. I would also maybe rename `events` to `records` as it's more representative of what these are.

Reply:
dig this and renaming to `records`

> previously `params` could be an empty string whereas now it would be `None`

the previous logic set `params` to empty string and `try/catch` parsed it to json, whereas this approach has no `try/catch` and would simply `json.dumps(d)`. so I don't think it'd be `None` and there's the potential risk of it throwing in that conversion?

Reply:
I think with this approach (ie a single for loop for logs), we'd need a try catch within the for loop in case `json.dumps` failed